jbmc/src/java_bytecode/java_bytecode_parse_tree.h

```diff
@@ -352,7 +352,8 @@ class fieldref_exprt : public exprt
 template <>
 inline bool can_cast_expr<fieldref_exprt>(const exprt &base)
 {
-  return base.get(ID_class) != ID_nil && base.get(ID_component_name) != ID_nil;
+  return base.get(ID_class) != ID_empty_string &&
```

Comment:
How about `return !base.get(ID_class).empty() && !base.get(ID_component_name).empty();`?

And while I'm at it: would mind adding a test case?

Reply:
Done